build.yaml

```diff
@@ -1007,143 +1008,7 @@ steps:
-    outputs:
-      - from: /io/test-output
-        to: /test-output-1
-    secrets:
-      - name: test-gsa-key
-        namespace:
-          valueFrom: default_ns.name
-        mountPath: /test-gsa-key
-    dependsOn:
-      - default_ns
-      - hail_run_tests_image
-      - build_hail
-  - kind: runImage
-    name: test_hail_java_2
-    image:
-      valueFrom: hail_run_tests_image.image
-    resources:
-      memory: standard
-      cpu: '2'
-    script: |
-      set -ex
-      cd /io
-      mkdir -p src/test
-      tar xzf resources.tar.gz -C src/test
-      tar xzf splits.tar.gz
-      export HAIL_TEST_SKIP_R=1
-      java -cp hail-test.jar:$SPARK_HOME/jars/* org.testng.TestNG -listener is.hail.LogTestListener testng-splits-2.xml
-    inputs:
-      - from: /resources.tar.gz
-        to: /io/resources.tar.gz
-      - from: /hail-debug-test.jar
-        to: /io/hail-test.jar
-      - from: /splits.tar.gz
-        to: /io/splits.tar.gz
-    outputs:
-      - from: /io/test-output
-        to: /test-output-2
-    secrets:
-      - name: test-gsa-key
-        namespace:
-          valueFrom: default_ns.name
-        mountPath: /test-gsa-key
-    dependsOn:
-      - default_ns
-      - hail_run_tests_image
-      - build_hail
-  - kind: runImage
-    name: test_hail_java_3
-    image:
-      valueFrom: hail_run_tests_image.image
-    resources:
-      memory: standard
-      cpu: '2'
-    script: |
-      set -ex
-      cd /io
-      mkdir -p src/test
-      tar xzf resources.tar.gz -C src/test
-      tar xzf splits.tar.gz
-      export HAIL_TEST_SKIP_R=1
-      java -cp hail-test.jar:$SPARK_HOME/jars/* org.testng.TestNG -listener is.hail.LogTestListener testng-splits-3.xml
-    inputs:
-      - from: /resources.tar.gz
-        to: /io/resources.tar.gz
-      - from: /hail-debug-test.jar
-        to: /io/hail-test.jar
-      - from: /splits.tar.gz
-        to: /io/splits.tar.gz
-    outputs:
-      - from: /io/test-output
-        to: /test-output-3
-    secrets:
-      - name: test-gsa-key
-        namespace:
-          valueFrom: default_ns.name
-        mountPath: /test-gsa-key
-    dependsOn:
-      - default_ns
-      - hail_run_tests_image
-      - build_hail
-  - kind: runImage
-    name: test_hail_java_4
-    image:
-      valueFrom: hail_run_tests_image.image
-    resources:
-      memory: standard
-      cpu: '2'
-    script: |
-      set -ex
-      cd /io
-      mkdir -p src/test
-      tar xzf resources.tar.gz -C src/test
-      tar xzf splits.tar.gz
-      export HAIL_TEST_SKIP_R=1
-      java -cp hail-test.jar:$SPARK_HOME/jars/* org.testng.TestNG -listener is.hail.LogTestListener testng-splits-4.xml
-    inputs:
-      - from: /resources.tar.gz
-        to: /io/resources.tar.gz
-      - from: /hail-debug-test.jar
-        to: /io/hail-test.jar
-      - from: /splits.tar.gz
-        to: /io/splits.tar.gz
-    outputs:
-      - from: /io/test-output
-        to: /test-output-4
+        to: /test-output
```

Comment:
Hmm this seems incorrect. All the test outputs will race to overwrite one another. I don’t think we care about the outputs though. Just remove them? Or add a template variable to outputs?

Reply:
It's not exactly explicit so happy to hear alternatives

Reply:
Ah. Hmm. A bit confusing. I don't have a better alternative though.